rocAL/rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -188,298 +211,478 @@ rocalFlip(
-    }
-    catch(const std::exception& e)
-    {
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
+    try {
+        RocalTensorlayout op_tensorLayout = (RocalTensorlayout)rocal_tensor_output_layout;
+        RocalTensorDataType op_tensorDataType = (RocalTensorDataType)rocal_tensor_output_datatype;
+        TensorInfo output_info = input->info();
+        output_info.set_tensor_layout(op_tensorLayout);
+        output_info.set_data_type(op_tensorDataType);
+        output = context->master_graph->create_tensor(output_info, is_output);
+        context->master_graph->add_node<SaturationNode>({input}, {output})->init(sat);
+    } catch(const std::exception& e) {
         context->capture_error(e.what());
         ERR(e.what())
     }
     return output;
 }
 
-RocalImage  ROCAL_API_CALL
-rocalCropResize(
-        RocalContext p_context,
-        RocalImage p_input,
-        unsigned dest_width, unsigned dest_height,
-        bool is_output,
-        RocalFloatParam p_area,
-        RocalFloatParam p_aspect_ratio,
-        RocalFloatParam p_x_center_drift,
-        RocalFloatParam p_y_center_drift)
-{
-    Image* output = nullptr;
-    if ((p_context == nullptr) || (p_input == nullptr)) {
-        ERR("Invalid ROCAL context or invalid input image")
-        return output;
-    }
 
-    auto context = static_cast<Context*>(p_context);
-    auto input = static_cast<Image*>(p_input);
+RocalTensor ROCAL_API_CALL
+rocalCropResize(RocalContext p_context,
+            RocalTensor p_input,
+            unsigned dest_width,
+            unsigned dest_height,
+            bool is_output,
+            RocalFloatParam p_area,
+            RocalFloatParam p_aspect_ratio,
+            RocalFloatParam p_x_center_drift,
+            RocalFloatParam p_y_center_drift,
+            RocalResizeScalingMode scaling_mode,
+            std::vector<unsigned> max_size,
+            unsigned resize_shorter,
+            unsigned resize_longer,
+            RocalResizeInterpolationType interpolation_type,
+            RocalTensorLayout rocal_tensor_output_layout,
+            RocalTensorOutputType rocal_tensor_output_datatype)
+{
+    Tensor* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
     auto area = static_cast<FloatParam*>(p_area);
     auto aspect_ratio = static_cast<FloatParam*>(p_aspect_ratio);
     auto x_center_drift = static_cast<FloatParam*>(p_x_center_drift);
     auto y_center_drift = static_cast<FloatParam*>(p_y_center_drift);
+
     try
     {
-        if(dest_width == 0 || dest_height == 0)
-            THROW("CropResize node needs tp receive non-zero destination dimensions")
-        // For the crop resize node, user can create an image with a different width and height
-        ImageInfo output_info = input->info();
+        if((dest_width | dest_height | resize_longer | resize_shorter) == 0)
```

Comment:
Why is the logical OR changed to bitwise OR operation here?

Reply:
Similar to resize, Suggested by Rajy for resize hence added similar changes here